pkg/instancegroups/settings_test.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instancegroups
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/kops/pkg/apis/kops"
+)
+
+func TestSettings(t *testing.T) {
+	for _, tc := range []struct {
+		name            string
+		defaultValue    interface{}
+		nonDefaultValue interface{}
+	}{
+		{
+			name:            "MaxUnavailable",
+			defaultValue:    intstr.FromInt(1),
+			nonDefaultValue: intstr.FromInt(2),
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			defaultCluster := &kops.RollingUpdate{}
+			setFieldValue(defaultCluster, tc.name, tc.defaultValue)
```

Comment:
I think I know where this is going... it is _usually_ easier and clearer just to write it out, but ... let's see once the chain lands!